src/main/kotlin/rx/lang/kotlin/observables.kt

```diff
@@ -1,12 +1,14 @@
 package rx.lang.kotlin
 
+import rx.AsyncEmitter
 import rx.Observable
 import rx.Subscriber
 import rx.Subscription
 import rx.observables.BlockingObservable
 
 fun <T> emptyObservable() : Observable<T> = Observable.empty()
 fun <T> observable(body : (s : Subscriber<in T>) -> Unit) : Observable<T> = Observable.create(body)
+fun <T> fromAsync(body: (s: AsyncEmitter<in T>) -> Unit, mode: AsyncEmitter.BackpressureMode): Observable<T> = Observable.fromAsync<T>(body, mode)
```

Comment:
I personally don't find that as a compelling argument. fromAsync by itself
offers no description as to what you are actually even creating for it to
be a top-level function by itself. Besides, what you describe is an IDE
deficiency, not a language problem to be solved by adding a new function.

Additionally, by that logic there should also be top-level functions for
just, from, concat, zip, etc. if no one wants to use imports which seems
ridiculous.

On Thu, Aug 18, 2016, 12:19 PM Ralph Bergmann <notifications@github.com>
wrote:

> In src/main/kotlin/rx/lang/kotlin/observables.kt
> <https://github.com/ReactiveX/RxKotlin/pull/78#discussion_r75339969>:
>
> >  import rx.Observable
> >  import rx.Subscriber
> >  import rx.Subscription
> >  import rx.observables.BlockingObservable
> >
> >  fun <T> emptyObservable() : Observable<T> = Observable.empty()
> >  fun <T> observable(body : (s : Subscriber<in T>) -> Unit) : Observable<T> = Observable.create(body)
> > +fun <T> fromAsync(body: (s: AsyncEmitter<in T>) -> Unit, mode: AsyncEmitter.BackpressureMode): Observable<T> = Observable.fromAsync<T>(body, mode)
>
> I know you prefer static imports but then I have to remember this import
> rule because there is no help from the IDE to add this import.
>
> With this method I have autocompletion from my IDE when I begin to enter
> fromAs.... It's just to make my work easier.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ReactiveX/RxKotlin/pull/78/files/253783be99b54149779b8d76cb57ab1e28959cf7#r75339969>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEEEYBUnEzKxfCwg0uBECVh46_tLhquks5qhIX3gaJpZM4JnpMR>
> .
>